lib/crawly/utils.ex

```diff
@@ -60,16 +73,30 @@ defmodule Crawly.Utils do
   def pipe(_, false, state), do: {false, state}
 
   def pipe([pipeline | pipelines], item, state) do
+    {module, args} =
```

Comment:
As I understand you're wrapping the Middlewares so they have a common look. {module, args}. I think it's better not to do it here. E.g. pipe should pipe. Not it wrapps and pipes :). Lets move the wrapping part away. Probably to the storage worker. Maybe here: https://github.com/oltarasenko/crawly/blob/ede6a6ebf39d63de79e8fc266bf80e07f1e79e39/lib/crawly/requests_storage/requests_storage_worker.ex#L65

Reply:
I don't quite understand what you mean. Are you referring to the naming use of `module` instead of `pipeline`? For this, part, I had thought that module was appropriate since both the middlewares and pipelines use `Util.pipe` and should behave the same way.

If I were to move the tuple-format checks to the request storage worker, I would have to duplicate the code to the item storage worker as well.

Reply:
Oh.  I did not think about Item pipelines. I general I would prefer pipe not to be responsible for wrapping/unwrapping. But, you're right. We don't have options atm :(